### PR TITLE
Python 3.x fixes and codejail config 

### PR DESCRIPTION
**Pull request by shaidar**
This PR addresses the following two issues:
- Reported issue where the learner input was not being properly tokenized and thus `import` statements weren't being read and causing an error to be thrown that no import statement are specified.
- When the `limits` object is set in the codejail configuration block it causes the command used for launching a watcher process to be invalid. This is because there was a variable name used in a for loop that shadowed a variable from an outer scope that is then returned from the `enable_codejail` method that is used as an argument for the execution.
- The PR that removed the usage of `six` failed to wrap the conditional `environment or {}` in parentheses before calling `.items()` on the resulting value. This would lead to trying to iterate over _either_ `environment` or `{}.items()` which is not the correct behavior.

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @shaidar! I've created [OSPR-5693](https://openedx.atlassian.net/browse/OSPR-5693) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by natabene** [comment]
@shaidar Thank you for your contribution. I am lining this up for our review.

**Comment by pdpinch** [comment]
Not sure, but I think this might have been a regression related to https://github.com/edx/xqueue-watcher/pull/56 

@UsamaSadiq or @iamsobanjaved can you take a look?

As far as I can tell, this code isn't covered by tests. We were going to hold off until we could get some tests written, but I just heard this is causing problems for a final exam that starts today.

**Comment by sarina** [comment]
@adzuci maybe? The two you've tagged are probably offline through Monday.

**Comment by pdpinch** [comment]
Thanks, I did mean #57. 

And I'm probably wrong about the cause the regression. It may be more of an environment issue -- I don't see how the code we're patching here could have ever worked, except that it wasn't touched in a Python 2.7

**Review by adzuci — APPROVED** [review]

**Comment by adzuci** [comment]
I think it would be better to merge the PR rather than https://github.com/edx/xqueue-watcher/pull/59/files, but since there is an exam going on until 3/30 and edx has rolled back to the version of the xqwatcher from 3/8 in prod I think it may be better to leave the service on python 2.7 until Monday.  

Note that this code likely won't get pulled in until we unpause our xqwatcher pipelines and do a deploy on python 3.5: https://github.com/mitodl/graders-mit-600x/blob/master/graders/python3graders

**Review by UsamaSadiq — APPROVED** [review]
Looks good to me. We can merge it and then try deploying to Python35 again once the test is over.

**Comment by UsamaSadiq** [comment]
@shaidar Thanks for this effort. Just a heads-up, before merging this PR, I'll squash the commits and rewrite the commit message so it follows the [Convention Commit guidelines.](https://github.com/edx/open-edx-proposals/blob/e2054a7c433170ec699f65c2ff235ef176dd259e/oeps/conventional-commits.rst)

File: grader_support/gradelib.py
Comment (doctoryes) [inline grader_support/gradelib.py:241]: This comment should also be removed, as the bug is fixed and the bug workaround is removed.

**Review by doctoryes — COMMENTED** [review]
Just a comment removal - otherwise 👍🏼

**Comment by adzuci** [comment]
@shaidar or @blarghmatey, we are planning to ship this this morning.  Would one of you be able to update the PR to resolve the comment by @doctoryes?

**Comment by edx-webhook** [comment]
@shaidar 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/xqueue-watcher&entry.867055334=https://github.com/edx/xqueue-watcher/pull/58&entry.1484655318=https://github.com/shaidar&entry.752974735=2021-03-26+14:13&entry.1917517419=2021-04-07+14:56&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**
